### PR TITLE
e2e: deflake hpa e2e custom metrics tests

### PR DESCRIPTION
**Pull request by pbetkier**
#### What type of PR is this?

/kind flake

#### What this PR does / why we need it:

Deflakes HPA e2e custom metrics tests:

a) adds namespacing to metrics: fixes interference between `should scale up when one metric is missing (Pod and External metrics)` and `should not scale down when one metric is missing (Container Resource and External Metrics)` specs, cause of flakiness ([example run](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm/1655105670397038592)).

b) replaces deployments containing unused exporters (metrics ignored) with deployments without any exporters: potential fix for often hitting a rate-limit on creating metrics descriptors ([429 errors, example run](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm/1655205579875946496)), also adds clarity.

c) fixes metric types: some external metrics tests used non-average type while expecting the value to be constant regardless of the number of pods. However, queries resulting from metric specs don't filter by pods, so a sum of metrics for all the pods is the fetched metric value (https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-metrics-not-related-to-kubernetes-objects). Adding averaging back by the number of pods fixes a couple of specs where the tests were passing for the wrong reason (wanted different test conditions).

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go [hunk 1]
```diff
@@ -637,3 +632,18 @@ func ensureDesiredReplicasInRange(ctx context.Context, deploymentName, namespace
...
 	framework.ExpectNoErrorWithOffset(1, err)
 }
+
+func noExporterDeployment(name, namespace string, replicas int32) *appsv1.Deployment {
+	d := e2edeployment.NewDeployment(name, replicas, map[string]string{"name": name}, "", "", appsv1.RollingUpdateDeploymentStrategyType)
+	d.ObjectMeta.Namespace = namespace
+	d.Spec.Template.Spec = v1.PodSpec{Containers: []v1.Container{
+		{
+			Name:            "sleeper",
+			Image:           "k8s.gcr.io/ubuntu-slim:0.1",
```
Comment (aojea) [inline test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go:642]: you have the agnhost image that is multiplatform and already has a pause command https://pkg.go.dev/k8s.io/kubernetes/test/images/agnhost#readme-pause

Comment (pbetkier) [reply test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go:642]: thanks!

**Comment by pbetkier** [comment]
Yea, I see. The presubmit got broken by #29019, I will work on fixing it first.

**Comment by aojea** [comment]
> Can't connect to 'docker' daemon.  please fix and retry.


there is a preset on the jobs to enable docker in docker

**Comment by aojea** [comment]
```
preset-dind-enabled: "true"
```

**Comment by pbetkier** [comment]
thanks! :)

/test pull-kubernetes-e2e-autoscaling-hpa-cm

**Comment by pbetkier** [comment]
yet another attempt at fixing presubmits...

/test pull-kubernetes-e2e-autoscaling-hpa-cm

**Comment by pbetkier** [comment]
/test pull-kubernetes-e2e-autoscaling-hpa-cm

retesting with fixes described above applied

**Comment by aojea** [comment]
I leave you an approve since the less I know of these tests looks correct, but it will be nice if someone more familiar can review for the final lgtm
/approve

**Pull request merged**